cli/installer/install-azd.sh

```diff
@@ -322,21 +330,49 @@ if [ "$platform" = "darwin" ] && [ "$skip_verify" = false ]; then
     fi
 fi
 
-install_location="$install_folder/azd"
-if [ -w "$install_folder/" ]; then
-    mv "$tmp_folder/$bin_name" "$install_location"
-else
+if [[ ! -d "$install_folder" ]]; then
+    say_verbose "Install folder does not exist: $install_folder. Creating..." 
+
+    if ! mkdir -p "$install_folder"; then 
+        say "Creating $install_folder requires elevated permission. You may be prompted to enter credentials." 
+        if ! sudo mkdir -p "$install_folder"; then
+            say_error "Could not create install folder: $install_folder"
+            save_error_report_if_enabled "InstallFailed" "SudoMkdirFailure"
+            exit 1
+        fi
+    fi
+fi
+
+mv_preface=""
+if [ ! -w "$install_folder/" ]; then
     say "Writing to $install_folder/ requires elevated permission. You may be prompted to enter credentials."
-    if ! sudo mv "$tmp_folder/$bin_name" "$install_location"; then
-        say_error "Could not copy file to install location: $install_location"
-        save_error_report_if_enabled "InstallFailed" "SudoMoveFailure"
+    mv_preface="sudo"
+fi
+if ! $mv_preface mv -f "$tmp_folder"/* "$install_folder"; then
+    say_error "Could not move files to install location: $install_folder"
+    save_error_report_if_enabled "InstallFailed" "SudoMoveFailure"
+    exit 1
+fi
+
+if [ "$symlink_folder" != "" ]; then
+    ln_preface=""
+    if [ ! -w "$symlink_folder/" ]; then
+        say "Writing to $symlink_folder/ requires elevated permission. You may be prompted to enter credentials." 
+        ln_preface="sudo"
+    fi
+    if ! $ln_preface ln -fs "$install_folder/$bin_name" "$symlink_folder/azd"; then
+        say_error "Could not create symlink to azd in $symlink_folder"
+        save_error_report_if_enabled "InstallFailed" "SymlinkCreateFailure"
         exit 1
     fi
 fi
 
 say_verbose "Cleaning up temp folder: $tmp_folder"
 rm -rf "$tmp_folder"
-say "Successfully installed to $install_location"
+say "Successfully installed to $install_folder"
+if [ "$symlink_folder" != "" ]; then
+    say "Symlinked in $symlink_folder/azd"
```

Comment:
nit: I almost expected "Symlinked to" instead of "Symlinked in", but I'm really unsure on terminology here.